src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -442,11 +441,11 @@ gl_sframe activity_classifier::predict_per_window(gl_sframe data,
 
   // Accumulate the class probabilities for each prediction window.
   gl_sframe raw_preds_per_window = perform_inference(data_it.get());
-
-  gl_sframe result =
-      gl_sframe({{"session_id", raw_preds_per_window["session_id"]},
-                 {"probability_vector", raw_preds_per_window["preds"]}});
-  result = result.add_row_number("prediction_id");
+  gl_sframe result = gl_sframe();
+  result.add_column(gl_sarray::from_sequence(0, raw_preds_per_window.size()),
+                    "prediction_id");
+  result.add_column(raw_preds_per_window["session_id"], "session_id");
+  result.add_column(raw_preds_per_window["preds"], "probability_vector");
```

Comment:
Could you explain this change?

Reply:
Wait a second, is this related to #1706? @shreyajain17 can you comment on what we should be doing here?